src/app/_types/sortTypes.ts

```diff
@@ -0,0 +1,17 @@
+export type SortOptionType = 'newest' | 'oldest'
+
+export type SortSetting = {
+  id: SortOptionType
+  name: string
+}
+
+export type SortableByDate = {
+  publishedOn?: string
+  startDate?: string
+}
+
+export type SortEntriesParams<T> = {
+  entries: T[]
+  dateField: 'startDate' | 'publishedOn'
```

Comment:
What about this instead?

```suggestion
  dateField: keyof SortableByDate
```